packages/ssr/src/createBrowserClient.ts

```diff
@@ -0,0 +1,114 @@
+import { createClient } from '@supabase/supabase-js';
+import { mergeDeepRight } from 'ramda';
+import { isBrowser } from './utils';
+import { parse, serialize } from 'cookie';
+
+import type { SupabaseClient } from '@supabase/supabase-js';
+import type {
+	GenericSchema,
+	SupabaseClientOptions
+} from '@supabase/supabase-js/dist/module/lib/types';
+import type { CookieOptionsWithName } from './types';
+
+let cachedBrowserClient: SupabaseClient<any, string> | undefined;
+
+export function createBrowserClient<
+	Database = any,
+	SchemaName extends string & keyof Database = 'public' extends keyof Database
+		? 'public'
+		: string & keyof Database,
+	Schema extends GenericSchema = Database[SchemaName] extends GenericSchema
+		? Database[SchemaName]
+		: any
+>(
+	supabaseUrl: string,
+	supabaseKey: string,
+	options: SupabaseClientOptions<SchemaName> & {
+		cookieOptions?: CookieOptionsWithName;
+		isSingleton?: boolean;
+	}
+) {
+	if (!supabaseUrl || !supabaseKey) {
+		throw new Error(
+			`Your project's URL and Key are required to create a Supabase client!\n\nCheck your Supabase project's API settings to find these values\n\nhttps://supabase.com/dashboard/project/_/settings/api`
+		);
+	}
+
+	const { isSingleton = true, cookieOptions, ...userDefinedClientOptions } = options;
+
+	const cookieClientOptions = {
+		global: {
+			headers: {
+				'X-Client-Info': `${PACKAGE_NAME}@${PACKAGE_VERSION}`
+			}
+		},
+		auth: {
+			flowType: 'pkce',
+			autoRefreshToken: isBrowser(),
+			detectSessionInUrl: isBrowser(),
+			persistSession: true,
+			storage: {
+				getItem: (key: string) => {
+					if (isBrowser()) {
+						const cookies = parse(document.cookie);
+						return cookies[key];
+					}
+
+					return null;
+				},
+				setItem: (key: string, value: string) => {
+					if (isBrowser()) {
+						document.cookie = serialize(key, value, {
+							httpOnly: false,
+							...cookieOptions
+						});
+					}
+				},
+				removeItem: (key: string) => {
+					if (isBrowser()) {
+						document.cookie = serialize(key, '', {
+							maxAge: 0,
+							httpOnly: false,
+							...cookieOptions
+						});
+					}
+				}
+			}
+		}
+	};
+
+	// Overwrites default client config with any user defined options
+	const clientOptions = mergeDeepRight(
```

Comment:
@silentworks this `mergeDeepRight` utility makes the default configuration above much more readable. What are your thoughts?